programs/locker/src/instructions/cancel_vesting_escrow.rs

```diff
@@ -0,0 +1,103 @@
+use anchor_spl::token::{Token, TokenAccount};
+
+use crate::*;
+use crate::util::token::transfer_to_recipient;
+
+/// Accounts for [locker::cancel_vesting_escrow].
+#[derive(Accounts)]
+#[event_cpi]
+pub struct CancelVestingEscrow<'info> {
+    /// Escrow.
+    #[account(mut)]
+    pub escrow: AccountLoader<'info, VestingEscrow>,
+
+    #[account(mut)]
+    pub escrow_token: Box<Account<'info, TokenAccount>>,
+
+    #[account(mut)]
+    pub creator_token: Box<Account<'info, TokenAccount>>,
+
+    #[account(mut)]
+    pub recipient_token: Box<Account<'info, TokenAccount>>,
+
+    /// Signer.
+    #[account(mut)]
+    pub signer: Signer<'info>,
+
+    /// Token program.
+    pub token_program: Program<'info, Token>,
+
+    /// System program.
+    pub system_program: Program<'info, System>,
+}
+
+pub fn handle_cancel_vesting_escrow(
+    ctx: Context<CancelVestingEscrow>
+) -> Result<()> {
+    let mut escrow = ctx.accounts.escrow.load_mut()?;
+    let escrow_token = anchor_spl::associated_token::get_associated_token_address(
+        &ctx.accounts.escrow.key(),
+        &escrow.token_mint,
+    );
+
+    require!(
+        escrow.cancelled == 0,
+        LockerError::AlreadyCancelled,
+    );
+
+    require!(
+        escrow_token == ctx.accounts.escrow_token.key(),
+        LockerError::InvalidEscrowTokenAddress
+    );
+
+    require!(
+        escrow.recipient == ctx.accounts.recipient_token.owner && escrow.token_mint == ctx.accounts.recipient_token.mint,
+        LockerError::InvalidRecipientTokenAccount,
+    );
+
+    require!(
```

Comment:
can we use macro check in context?

Reply:
fixed, move the validation inside `impl<'info> CancelVestingEscrow<'info> {`